test/Microsoft.TestPlatform.Common.UnitTests/Filtering/FastFilterTests.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.TestPlatform.Common.UnitTests.Filtering
+{
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Microsoft.VisualStudio.TestPlatform.Common.Filtering;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+
+    [TestClass]
+    public class FastFilterTests
```

Comment:
Can we have a test for something like this too please: `FullyQualifiedName=a|(FullyQualifiedName=b|FullyQualifiedName=c)`?